File: typescript/prototype/manager/Manager.test.ts
import { assertEquals } from "https://deno.land/std@0.81.0/testing/asserts.ts";
import { Manager } from "./Manager.ts";
import { Product } from "./Product.ts";

Deno.test({
  name:
    "Manager.create() should retuen the same class with registered products.",
  fn(): void {
    const sut = new Manager();
    const mock: Product = {
      use: (msg: string) => msg,
    };
    sut.register("dummy", mock);
    const actual = sut.create("dummy")
    assertEquals(actual.use("Hello"), "Hello");
  },
});

File: go/prototype/manager/manager_test.go
package manager_test

import (
	"testing"

	"github.com/hitsuji-haneta/design-pattern/go/prototype/box"
	"github.com/hitsuji-haneta/design-pattern/go/prototype/manager"
	. "github.com/onsi/ginkgo"
	. "github.com/onsi/gomega"
)

func TestVisualizer(t *testing.T) {
	RegisterFailHandler(Fail)
	RunSpecs(t, "Go Suite")
}

var _ = Describe("Manager", func() {
	Context(".Create", func() {

		It("should return a clone of product", func() {
			sut := manager.New()

			starbox := box.New("*")
			sut.Register("star box", starbox)
			actual := sut.Create("star box")
			Expect(actual).To(Equal(starbox))
		})
	})
})
